### ISSUE TITLE
Graph stacking should automatically consider NULL as 0

### ISSUE THREAD

**Issue by sylr** [grafana/grafana#15690]
I know there as been a number of issues about this but using `null as 0` is not a good solution because it messes up with the aggregation values (min, max, average, current) in the legend.

If you are stacking you are obliged to use `null as 0` otherwise you might have surprises in your graphs but if you do so you can't trust the aggregation values in the legend because a Null in the timerange will:

- Set min to 0 even if your metric never had this value
- Set current to 0 even if the metric (/ label combination) does not exist anymore
- And it will make the average totally bogus

So my proposal is to automatically consider Null values as being 0 for the graphic representation when stacking so that you can choose whether or not you want to consider Null as 0 for the aggregation functions.

Cheers.

**Comment by xlson** [comment]
Leaving this one to you @torkelo @Ijin08

**Comment by roidelapluie** [comment]
I agree with OP.

**Comment by sylr** [comment]
@torkelo @Ijin08 care to comment on this ?

**Comment by sylr** [comment]
@torkelo @Ijin08 ?

**Comment by onumossn** [comment]
Any plans for this?

**Comment by grandich** [comment]
+1

I'm new to Grafana and painfully stumbled upon this. One example of 'surprises' in graphs: when turning on and off Prometheus node exporter' collectors:

With Null Value=null
![image](https://user-images.githubusercontent.com/1310215/112666637-27fc6880-8e3b-11eb-9bc1-7ca12a8df2fc.png)

With Null Value=null as zero
![image](https://user-images.githubusercontent.com/1310215/112666666-2cc11c80-8e3b-11eb-8a50-f863781a0ab3.png)

Albeit no difference in legends...

**Comment by djahandarie** [comment]
I think this is a UI/readability issue pretty much anyone who uses stacked graphs in Grafana will run into. Is there any chance this can get prioritized?

**Comment by jhoos70** [comment]
Me too, missing the possibility to keep zero values down at the axis.

**Comment by tvsaru** [comment]
I am yet another person who tripped on this weird behavior. 

In my example screenshot below, of stacking graphs, the yellow line is supposed to be 0, while green line was 1, but the yellow line masks the green line giving a wrong visualization of the data (makes it look like yellow was 1 for the whole time period).

![image](https://user-images.githubusercontent.com/11013485/195492729-f71f6a73-d97e-42e0-b98e-bee8e8ba85e6.png)

**Comment by AndrewJDR** [comment]
Looking forward to a solution for this as well

**Comment by sylr** [comment]
@torkelo @gelicia @Ijin08 could you provide an update on this ?

**Comment by richvdh** [comment]
This is still an issue as far as I am aware

**Comment by jhoos70** [comment]
Yes, the problem still exists and makes stacking in many cases senseless.
